plugins/scaffolder/src/hooks/useStarredEntities.test.tsx

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { PropsWithChildren } from 'react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useStarredEntities } from './useStarredEntities';
+import {
+  ApiProvider,
+  ApiRegistry,
+  storageApiRef,
+  WebStorage,
+  StorageApi,
+} from '@backstage/core';
+import { MockErrorApi } from '@backstage/test-utils';
+import { Entity } from '@backstage/catalog-model';
+
+describe('useStarredEntities', () => {
+  let mockStorage: StorageApi | undefined;
+
+  const mockEntity: Entity = {
+    apiVersion: '1',
+    kind: 'Component',
+    metadata: {
+      name: 'mock',
+    },
+  };
+
+  const secondMockEntity: Entity = {
+    apiVersion: '1',
+    kind: 'Component',
+    metadata: {
+      namespace: 'test',
+      name: 'mock2',
+    },
+  };
+
+  const wrapper = ({ children }: PropsWithChildren<{}>) => {
+    return (
+      <ApiProvider apis={ApiRegistry.with(storageApiRef, mockStorage)}>
+        {children}
+      </ApiProvider>
+    );
+  };
+
+  beforeEach(() => {
+    mockStorage = new WebStorage('@backstage', new MockErrorApi()).forBucket(
+      Date.now().toString(), // TODO(blam): need something that changes every test run for now until the MockStorage is implemented
+    );
+  });
+  it('should return an empty set for when there is no items in storage', async () => {
+    const { result } = renderHook(() => useStarredEntities(), { wrapper });
+
+    expect(result.current.starredEntities.size).toBe(0);
+  });
+  it('should return a set with the current items when there is items in storage', async () => {
+    const expectedIds = ['i', 'am', 'some', 'test', 'ids'];
+    const store = mockStorage?.forBucket('settings');
+    await store?.set('starredEntities', expectedIds);
+
+    const { result } = renderHook(() => useStarredEntities(), { wrapper });
+
+    for (const item of expectedIds) {
+      expect(result.current.starredEntities.has(item)).toBeTruthy();
+    }
+  });
+  it('should listen to changes when the storage is set elsewhere', async () => {
+    const { result, waitForNextUpdate } = renderHook(
+      () => useStarredEntities(),
+      { wrapper },
+    );
+
+    expect(result.current.starredEntities.size).toBe(0);
+    expect(result.current.isStarredEntity(mockEntity)).toBeFalsy();
+
+    // Make this happen after awaiting for the next update so we can
+    // catch when the hook re-renders with the latest data
+    setTimeout(() => result.current.toggleStarredEntity(mockEntity), 1);
+
+    await waitForNextUpdate();
+
+    expect(result.current.starredEntities.size).toBe(1);
+    expect(result.current.isStarredEntity(mockEntity)).toBeTruthy();
+  });
+
+  it('should write new entries to the local store when adding a togglging entity', async () => {
```

Comment:
```suggestion
  it('should write new entries to the local store when adding a toggling entity', async () => {
```

Reply:
Fixed in 452602094f23e5bb0eb150e649c9661066d2f75e